src/console/Startup.cs

```diff
@@ -188,5 +194,28 @@ public string GetConsoleApplicationType()
 
             return applicationType;
         }
+
+        private void AddContentTemplateFactories(IServiceCollection services)
+        {
+            services.AddSingleton<IExpressionRegister>(sp => new AssemblyExpressionRegister(typeof(IExpressionRegister).Assembly, sp.GetRequiredService<ITelemetryLogger>()));
+            services.AddSingleton<IExpressionEvaluatorFactory>(
+                sp =>
+                {
+                    var jmesPath = new JmesPath();
+                    var expressionRegister = sp.GetRequiredService<IExpressionRegister>();
+                    expressionRegister.RegisterExpressions(jmesPath.FunctionRepository);
+                    return new TemplateExpressionEvaluatorFactory(jmesPath);
+                });
+            services.AddSingleton<ITemplateFactory<TemplateContainer, IContentTemplate>, JsonPathContentTemplateFactory>();
+            services.AddSingleton<ITemplateFactory<TemplateContainer, IContentTemplate>, IotJsonPathContentTemplateFactory>();
+            services.AddSingleton<ITemplateFactory<TemplateContainer, IContentTemplate>, IotCentralJsonPathContentTemplateFactory>();
+            services.AddSingleton<ITemplateFactory<TemplateContainer, IContentTemplate>, CalculatedFunctionContentTemplateFactory>();
+            services.AddSingleton<CollectionTemplateFactory<IContentTemplate, IContentTemplate>>(
+                sp =>
+                {
+                    var contentTemplateFactories = sp.GetRequiredService<IEnumerable<ITemplateFactory<TemplateContainer, IContentTemplate>>>();
+                    return new CollectionContentTemplateFactory(contentTemplateFactories.ToArray());
```

Comment:
If you add the ctor for IEnumerable to CollectionContentTemplateFactory you can avoid this here.